src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -371,6 +374,96 @@ def image_mask_iterator():
     return nwbfile
 
 
+def add_fluorescence(segmentation_extractor: SegmentationExtractor, nwbfile: NWBFile, metadata: dict) -> NWBFile:
+    """
+    Adds the fluorescence specificied by the metadata to the nwb file.
+    The fluorescence that is added is the one located in metadata["Ophys"]["Fluorescence"]
+
+    Parameters
+    ----------
+    nwbfile : NWBFile
+        An previously defined -in memory- NWBFile.
+    metadata : dict
+        The metadata in the nwb conversion tools format.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    ophys = get_module(nwbfile, "ophys")
+    if fluorescence_name in ophys.data_interfaces:
+        return nwbfile
+
+    add_plane_segmentation(segmentation_extractor=segmentation_extractor, nwbfile=nwbfile, metadata=metadata)
+
+    # Get plane segmentation from the image segmentation
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    image_segmentation_name = image_segmentation_metadata["name"]
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    plane_segmentation_name = image_segmentation_metadata["plane_segmentations"][0]["name"]
+    plane_segmentation = image_segmentation.plane_segmentations[plane_segmentation_name]
+
+    # TODO what should be done with plane_no_loop?
+    plane_no_loop = 0
+    # Create a reference for ROIs from the plane segmentation
+    roi_table_region = plane_segmentation.create_roi_table_region(
+        region=segmentation_extractor.get_roi_ids(),
+        description=f"region for Imaging plane{plane_no_loop}",
+    )
+
+    # Get traces from the segmentation extractor
+    roi_response_dict = segmentation_extractor.get_traces_dict()
+
+    # Filter empty data
+    roi_response_dict = {key: value for key, value in roi_response_dict.items() if value is not None}
+    # Filter all zero data
+    roi_response_dict = {key: value for key, value in roi_response_dict.items() if np.any(value)}
```

Comment:
This is a neat trick for `np.any`!

I was wondering on how does this compare against the previous generator solution proposed by @CodyCBakerPhD in terms of speed and I got the following:

```
n [11]: timeit.timeit("np.any(np.ones(int(1e6)))", setup="import numpy as np",
    ...: number=100)
Out[11]: 0.1355249520020152

In [12]: timeit.timeit("any(x!=0 for x in np.ones(int(1e6)))", setup="import num
    ...: py as np", number=100)
Out[12]: 0.0691819530002249
```

So it seems that if you expected the traces to have mostly non-zero data most of the time then @CodyCBakerPhD solution seems better. However, if a lot of the times the tested vectors are going to have mostly zeros we have the opposite result:

```
In [16]: timeit.timeit("np.any(np.zeros(int(1e6)))", setup="import numpy as np",
    ...:  number=100)
Out[16]: 0.11136360900127329

In [17]: timeit.timeit("any(x!=0 for x in np.zeros(int(1e6)))", setup="import nu
    ...: mpy as np", number=100)
Out[17]: 21.058059448998392
```
Where we can see that the generator solution does in fact quite bad. So I don't know what gives, what do you guys think?

Reply:
It's true, all that logic in numpy is vectorized so it's ideal for algebraic operations on large arrays that are contiguous in memory, but has no early stop condition. So lazy early return generators will behave better whenever the non-zero data density is somewhat high, and arrays will be better for when the non-zero data is very sparse.

Also the generator is lazy in memory so if the `value` of that response series is very large we essentially have iterative read taken care of by default whereas numpy would load it all in at once - how large are these data usually? I guess it may not be too bad (`frames x number_of_rois` I'd guess?). I'd think this is actually the more important consideration and the early return is just a bonus.

So which is better for speed depends on the natural statistics of this application - are the response series we want to ignore predominantly zero with sparse non-zero entries or are they pretty densely populated with non-zero? I guess either way the worst case time for a full array of zero makes numpy the preferred approach dependent on the memory consideration above.

Reply:
I thought about the lazy point as well but right now in roiextractor the response series (traces) are already preloaded numpy arrays so I don't think the point applies yet. I agree that if at some point the return of the series are memmaps or any other lazyly loaded object and the generator reads bit by bit then that would be the most important consideration.

Reply:
Ah, fair enough then. I'd have thought it was possible (for some of our current conversions at least) to treat the segmentation data lazily

Reply:
What you say would work best in the ideal case though. I think that's a good point. Let's see what @weiglszonja thinks.

Reply:
Thank you for doing the comparisons @h-mayorquin, I expected the numpy approach to be faster for arrays where the non-zero data is sparse, I did not consider the lazy point though. Thanks @CodyCBakerPhD for explaining it. I guess most of the time we would not expect nearly all-zero traces, then the generator approach is way better. I'd go with that approach here instead of numpy.

Reply:
One more suggestion over here, maybe it would be good to change the key: value variables to something more descriptive? `trace_name` and `trace` is what comes to mind but if you come with something better that's great.